addons/mrp/models/mrp_production.py

```diff
@@ -2003,10 +2003,10 @@ def action_merge(self):
         for move in self.move_finished_ids:
             dests.setdefault(move.byproduct_id.id, []).extend(move.move_dest_ids.ids)
 
-        production = self.env['mrp.production'].create({
+        production = self.env['mrp.production'].with_context(default_picking_type_id=self.picking_type_id.ids[0]).create({
```

Comment:
To get the default location from picking we set this context
this is done here https://github.com/odoo/odoo/blob/saas-15.2/addons/mrp/models/mrp_production.py#L43

Reply:
That's a good point. I will check what makes the most sense in this case since the source and destination locations can be changed in the MO to not match the picking_type default locations